methods/basic.go

```diff
@@ -15,7 +20,62 @@ func BasicRequestWithBody(c *cli.Context, method string) (string, error) {
 		return "", err
 	}
 
-	var jsonStr = []byte(c.String("body"))
+	var jsonStr []byte
+
+	if c.Bool("editor") {
+		var path string
+		var editor string
+
+		// check OS
+		currentOs := runtime.GOOS
+
+		// based on OS find assign default editor
+		if currentOs == "linux" || currentOs == "darwin" {
+			editor = "nano"
```

Comment:
It's better to respect `$EDITOR` on linux and macOS. You can check if it is set, and if not, default to nano. By default on, eg, Ubuntu, it is not set, so if someone has set it, then it is an explicit preference.

`os.Getenv("EDITOR")` will return an empty string if it is not set.

Something like:

```go
editor = os.Getenv("EDITOR")
if editor == "" {
    editor = "nano"
}
```